sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/ShowPartitionsSuite.scala

```diff
@@ -93,4 +93,26 @@ class ShowPartitionsSuite extends ShowPartitionsSuiteBase with CommandSuiteBase
       assert(sql("SHOW PARTITIONS part_datasrc").count() == 3)
     }
   }
+
+  test("null and empty string as partition values") {
+    import testImplicits._
+    withTable("t") {
+      val df = Seq(
+        (0, ""),
+        (1, null)
+      ).toDF("a", "part")
+      df.write
+        .partitionBy("part")
+        .format("parquet")
+        .mode(SaveMode.Overwrite)
+        .saveAsTable("t")
+
+      runShowPartitionsSql(
+        "SHOW PARTITIONS t",
+        Row("part=__HIVE_DEFAULT_PARTITION__") :: Nil)
```

Comment:
This is a bug... We shouldn't expose the internal stuff. We should return null as `df.show` did. Let's fix it later.

Reply:
ah, if the behavior is the same as Hive, we may need to keep  it.